### DISCUSSION TITLE
Explore: Allow ad hoc filter from log row detail for Elastic and Influxdb

### DISCUSSION THREAD

**Discussion by marefr** [grafana/grafana#45683] [Observability]
**What would you like to be added**:
A general way to show meta data of a log message in Explore and being able to filter log messages by adding filters to query field from a log row (similar to ad-hoc filter support in dashboards). The primary way to filter logs result should be by clicking on a value (label/tag/field) on a log row rather than manually input to/select values in query field.

**Why is this needed**:
Loki is currently the only datasource that supports showing labels when visualizing logs in Explore, see screenshot below:
![image](https://user-images.githubusercontent.com/1668778/60356530-86c85000-99d1-11e9-9750-c5ed1379c1ca.png)

To help users get more context when looking at a log row and more easily filter their logs we need to find a  general way to support more logs-datasources than Loki.

Additionally, this should support filtering for (matches) and out (not matches) values to make it work very similar to existing ad-hoc filters in dashboards.

**Comment by jcdauchy** [comment]
Is it possible to use influxdb as a log datasource in Explore / Logpanel ? If this is the case, any input how to set this up ?

**Comment by aocenas** [comment]
@jcdauchy Yes, there were some bugs that were fixed in 6.6 so it may be an issue of you hitting those bugs. Please try now, you should not need any special setup. If it does not work please file a separate issue.

**Comment by aocenas** [comment]
@marefr As we have the log row detail UI can we close this? I think that addresses this issue.

**Comment by marefr** [comment]
@aocenas I guess, but adhoc filtering for elasticsearch/infludb logs are yet not supported which was part of this. Either close this and open a new one or keep this open and change the title maybe?

**Comment by aocenas** [comment]
@marefr Ah you are right it work in loki only at the moment. Will change the title.

**Comment by gabor** [comment]
this works now in main-branch grafana. should be available in grafana9.4.0
